nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/resources/PropertiesBagResource.java

```diff
@@ -189,27 +190,33 @@ protected ActionReportResult clearThenSaveProperties(List<Map<String, String>> p
 
                 // the prop name can not contain .
                 // need to remove the . test when http://java.net/jira/browse/GLASSFISH-15418  is fixed
-                boolean canSaveDesc = !((Object)property.get("name")).toString().contains(".");
+                boolean isDottedName = ((Object)property.get("name")).toString().contains(".");
 
                 if ((existingProp == null) || !unescapedValue.equals(existingProp.getValue())) {
+                    
                     data.put(escapedName, ((Object)property.get("value")).toString());
-                    if (canSaveDesc && (description != null)) {
-                        data.put(escapedName + ".description", ((Object) description).toString());
+                    
+                    if (isDottedName) {
+                        data.put(unescapedName + ".name", unescapedName);
                     }
+                    
+                    if ((description != null)) {
```

Comment:
There are also lots of unnecessary parenthesis in this method, basically no expression needs them (since casts will be removed).